### PR TITLE
BatchNorm & RMSProp

### PR DESCRIPTION
**Pull request by cjacoby**
* Adding 'batch_norm' parameter to configs to allow you to include batch norm on all layers with a nonlinearity.
* Changing updates to use RMSProp per @bmcfee's recommendation.

Closes Issue #22 .

---
This change is [Reviewable](https://reviewable.io/reviews/ejhumphrey/harmonic-cnn/29)

### CODE REVIEW

**Comment by bmcfee** [comment]
Looks like I'm assigned to this -- what's needed?

**Comment by cjacoby** [comment]
Sign off/lgtm, I guess? Just got back from my trip this weekend; I'll be
hacking on this a bunch this coming week/weekend.

**Comment by bmcfee** [comment]
Ok.  Are we just not worrying about travis at this point?  Or should we try and resolve that before making any more merges?

File: wcqtlib/train/models.py [hunk 1]
```diff
@@ -156,7 +156,15 @@ def _build_network(self):
                 raise InvalidNetworkDefinition(
                     "Each layer must contain a 'type'")
 
-            network = layer_class(network, **layer)
+            if object_definition.get("batch_norm", False) is True \
+                    and 'nonlinearity' in layer:
+                network = lasagne.layers.batch_norm(
```
Comment (bmcfee) [inline wcqtlib/train/models.py]: I thought lasagne's batch norm automatically does the nonlinearity detection logic for us?  Or am I mis-remembering?

Also, backslashing across lines is to be discouraged: use parentheses instead.

Comment (cjacoby) [inline wcqtlib/train/models.py]: Ah, you are correct. Fixed.

**Comment by cjacoby** [comment]
Re: travis. I could go either way; it requires some changes from the minst dataset repo. I think I'll put some energy in over there, since we haven't heard anything from @ejhumphrey yet; as soon as we have that straightened out, it should be easy to fix travis here.

**Comment by bmcfee** [comment]
Sounds good.  Otherwise, this one LGTM.

**Comment by ejhumphrey** [comment]
hey @cjacoby what's the status of this dealy here? should be clear now that we've got minst mostly straightened out (and it looks like you added the tinyset to master?)

**Comment by cjacoby** [comment]
I believe this is done, but I have a branch cleaning up some other things
that are different now because of MINST. Going to try to finish that up
this morning, and then I should be able to close this guy.
